src/librustc_resolve/lib.rs

```diff
@@ -1110,6 +1140,11 @@ pub struct Resolver<'a, 'tcx:'a> {
 
     used_imports: HashSet<(NodeId, Namespace)>,
     used_crates: HashSet<CrateNum>,
+
+    // Callback function for intercepting walks
```

Comment:
It's to allow the walker to fall out while preserving the ribs (to call resolve_path afterwards). I've added a few more comments now.